db/db.go

```diff
@@ -0,0 +1,130 @@
+package db
+
+import (
+	"context"
+
+	sq "github.com/Masterminds/squirrel"
+	"github.com/jackc/pgx/v4"
+	"github.com/jackc/pgx/v4/pgxpool"
+)
+
+// DB database
+type DB struct {
+	pool *pgxpool.Pool
+	psql sq.StatementBuilderType
+}
+
+// New creates a new DB connection
+func New(pool *pgxpool.Pool, psql sq.StatementBuilderType) *DB {
+	return &DB{
+		pool: pool,
+		psql: psql,
+	}
+}
+
+// Post ...
+type Post struct {
+	ID    int    `json:"id"`
+	Title string `json:"title"`
+	Body  string `json:"body"`
+}
+
+// GetPosts fetches all posts
+func (db *DB) GetPosts() ([]Post, error) {
+	posts := []Post{}
+
+	sql, _, err := db.psql.Select("*").From("posts").ToSql()
+	if err != nil {
+		return []Post{}, err
+	}
+
+	rows, err := db.pool.Query(context.Background(), sql)
+	if err != nil {
+		return []Post{}, err
+	}
+
+	defer rows.Close()
```

Comment:
I know you'll see this used like this in a lot of examples, but it's actually a bit of a bad practice, because it ignores the error returned from `rows.Close`. A better solution is to make sure to check the error in a defer and assign to a _named return_ error:
```suggestion
	defer func() {
		err := rows.Close()
		if err != nil {
			retErr = err
		}
	}()
```

This requires changing the signature of the function too:
```go
func (db *DB) GetPosts() (_ ]Post, retErr error) {
```

I recommend reading point 3 in this blog for a complete example: https://blog.learngoprogramming.com/5-gotchas-of-defer-in-go-golang-part-iii-36a1ab3d6ef1